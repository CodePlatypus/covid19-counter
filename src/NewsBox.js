import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export default function NewsBox(props) {
    if (props.isLoading) {
        return (
            <div>
            </div>
        )
    }
    return (
        <BoxWrapper>
            <div>
                <a href={props.newsItem.link} target="_blank" rel="noopener noreferrer">
                    <NewsTitle>{props.newsItem.title}</NewsTitle>
                    {props.newsItem.description}
                <hr />
                </a>
            </div>
            
        </BoxWrapper>
    )
}

const BoxWrapper = styled.section`
  cursor: pointer;
  padding: 10px 0 0 0;
  &:hover {
    background-color: #f2f2f2;
  }
  a {
    color: black;
    text-decoration: none; /* no underline */
  }
`;

const NewsTitle = styled.section`
  font-weight: bold;
`;

NewsBox.propTypes = {
    isLoading: PropTypes.bool.isRequired,
    newsItem: PropTypes.object
}