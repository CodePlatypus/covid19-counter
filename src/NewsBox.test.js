import React from 'react';
import {shallow} from 'enzyme';
import NewsBox from './NewsBox';
import toJson from 'enzyme-to-json';

test('renders NewsBox correctly with newsItem isNotLoading', () => {
    const wrapper = shallow(<NewsBox isLoading={false}
                                     newsItem={newsItem} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test('renders NewsBox correctly with newsItem isLoading', () => {
    const wrapper = shallow(<NewsBox isLoading={true}
                                     newsItem={newsItem} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

const newsItem = {
    "title": "Title1",
    "description": "description1",
    "link": "https://link1.com"
};