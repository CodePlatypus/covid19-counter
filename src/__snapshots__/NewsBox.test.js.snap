// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NewsBox correctly with newsItem 1`] = `<div />`;

exports[`renders NewsBox correctly with newsItem isLoading 1`] = `<div />`;

exports[`renders NewsBox correctly with newsItem isNotLoading 1`] = `
<styled.section>
  <div>
    <a
      href="https://link1.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <styled.section>
        Title1
      </styled.section>
      description1
      <hr />
    </a>
  </div>
</styled.section>
`;
